core/src/main/scala/pureconfig/backend/ConfigFactoryWrapper.scala

```diff
@@ -22,15 +22,17 @@ object ConfigFactoryWrapper {
     unsafeToEither(ConfigFactory.load())
 
   /** @see `com.typesafe.config.ConfigFactory.parseFile()` */
-  def parseFile(path: Path): Either[ConfigReaderFailures, Config] =
-    unsafeToEither(ConfigFactory.parseFile(path.toFile))
+  def parseFile(path: Path): Either[ConfigReaderFailures, Config] = {
+    if (!path.toFile.isFile || !path.toFile.canRead) Left(ConfigReaderFailures(CannotReadFile(path)))
```

Comment:
I don't believe there's a functional difference, but if we're already using NIO, it makes sense to keep using it. It also avoids some extra allocations.